producer/src/main/kotlin/streams/kafka/KafkaEventRouter.kt

```diff
@@ -22,24 +22,26 @@ import java.util.concurrent.ThreadLocalRandom
 class KafkaEventRouter: StreamsEventRouter {
     private val log: Log
     private lateinit var producer: Producer<String, ByteArray>
-    private val kafkaConfig: KafkaConfiguration
+    private lateinit var kafkaConfig: KafkaConfiguration
 
 
     constructor(logService: LogService, config: Config): super(logService, config) {
         log = logService.getUserLog(KafkaEventRouter::class.java)
-        log.info("Initializing Kafka Connector")
-        kafkaConfig = KafkaConfiguration.from(config.raw)
     }
 
     override fun start() {
+        log.info("Initializing Kafka Connector")
+        kafkaConfig = KafkaConfiguration.from(config?.raw ?: emptyMap())
         val props = kafkaConfig.asProperties()
         producer = Neo4jKafkaProducer<String, ByteArray>(props)
         producer.initTransactions()
         log.info("Kafka Connector started")
     }
 
     override fun stop() {
-        producer.close()
+        if (::producer.isInitialized) {
```

Comment:
same here with the static initialization.

Reply:
ok